app/controlplane/internal/policies/policyprovider.go

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policies
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+// PolicyProvider represents an external policy provider
+type PolicyProvider struct {
+	name, host string
+	isDefault  bool
+}
+
+type ProviderResponse struct {
+	Policy map[string]any `json:"policy"`
+}
+
+// Resolve calls the remote provider for retrieving a policy
+func (p *PolicyProvider) Resolve(policyName string, token string) (*schemaapi.Policy, error) {
```

Comment:
let's add some simple validations that make sure that `name` and `token` are provided.